R/spaces_left_parentheses_linter.R

```diff
@@ -2,6 +2,19 @@
 #'
 #' Check that all left parentheses have a space before them unless they are in a function call.
 #'
+#' @examples
+#' # will produce lints
+#' lint(
```

Comment:
in an earlier linter (function_left_parentheses_linter? is that the name?) we showed if/function signature examples, should we do the reverse here?

Reply:
Not sure what you mean by "reverse" here.

Reply:
Complementary examples to these:

https://github.com/r-lib/lintr/blob/615f72adc8f299d1265a072e5dbe21be8486f120/R/function_left_parentheses_linter.R#L21-L40

Reply:
So including the okay examples but with a space to show that this doesn't lint with spaces_left linter?

``` r
library(lintr)

lint(
  text = "mean (x)",
  linters = spaces_left_parentheses_linter()
)

lint(
  text =  "stats::sd (c (x, y, z))",
  linters = spaces_left_parentheses_linter()
)

lint(
  text =  "if (TRUE) x else y",
  linters = spaces_left_parentheses_linter()
)

lint(
  text =  "foo <- function (x) (x + 1)",
  linters = spaces_left_parentheses_linter()
)
```

<sup>Created on 2022-10-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Hmm, now I'm doubting everything :)

I definitely don't think examples like `mean (x)` are helpful... but maybe the better thing to do is just remove the `if()` example from `function_left_parentheses_linter()`? Since `if()` is irrelevant to that linter, and is covered here instead.

And then add cross-links between these two linters' help pages since they're so closely related.

WDYT?

Reply:
SGTM. I have made the necessary changes. See if this is better.